connectors/src/connectors/zendesk/temporal/activities.ts

```diff
@@ -51,50 +71,157 @@ export async function saveZendeskConnectorSuccessSync({
+    if (brandInDb.ticketsPermission !== "read") {
+      await brandInDb.remove({ dataSourceConfig, loggerArgs });
+      return { helpCenterAllowed: false, ticketsAllowed: false };
+    }
+    await brandInDb.update({ helpCenterPermission: "none" });
+  }
+
+  // otherwise, we update the brand name and lastUpsertedTs
+  await brandInDb.update({
+    name: fetchedBrand.name || "Brand",
+    lastUpsertedTs: new Date(currentSyncDateMs),
+  });
+  return {
+    helpCenterAllowed: brandInDb.helpCenterPermission === "read",
+    ticketsAllowed: brandInDb.ticketsPermission === "read",
+  };
 }
 
 /**
  * This activity is responsible for checking the permissions for a Brand's Help Center.
  *
  * @returns true if the Help Center has read permissions enabled.
  */
-// eslint-disable-next-line no-empty-pattern
-export async function checkZendeskHelpCenterPermissionsActivity({}: {
+export async function checkZendeskHelpCenterPermissionsActivity({
+  connectorId,
+  brandId,
+}: {
   connectorId: ModelId;
   brandId: number;
 }): Promise<boolean> {
-  return false;
+  const brandInDb = await ZendeskBrandResource.fetchByBrandId({
+    connectorId,
+    brandId,
+  });
+  if (!brandInDb) {
+    throw new Error(
+      `[Zendesk] Brand not found, connectorId: ${connectorId}, brandId: ${brandId}`
+    );
+  }
+
+  return brandInDb.helpCenterPermission === "read";
 }
 
 /**
  * This activity is responsible for checking the permissions for a Brand's Tickets.
  *
  * @returns true if the Help Center has read permissions enabled.
  */
-// eslint-disable-next-line no-empty-pattern
-export async function checkZendeskTicketsPermissionsActivity({}: {
+export async function checkZendeskTicketsPermissionsActivity({
+  connectorId,
+  brandId,
+}: {
   connectorId: ModelId;
   brandId: number;
 }): Promise<boolean> {
-  return false;
+  const brandInDb = await ZendeskBrandResource.fetchByBrandId({
+    connectorId,
+    brandId,
+  });
+  if (!brandInDb) {
+    throw new Error(
+      `[Zendesk] Brand not found, connectorId: ${connectorId}, brandId: ${brandId}`
+    );
+  }
+
+  return brandInDb.ticketsPermission === "read";
 }
 
 /**
  * Retrieves the categories for a given Brand.
  */
-// eslint-disable-next-line no-empty-pattern
-export async function getZendeskCategoriesActivity({}: {
+export async function getZendeskCategoriesActivity({
+  connectorId,
+  brandId,
+}: {
   connectorId: ModelId;
   brandId: number;
 }): Promise<number[]> {
-  return [];
+  const connector = await _getZendeskConnectorOrRaise(connectorId);
+  const configuration = await _getZendeskConfigurationOrRaise(connectorId);
+  const accessToken = await getZendeskAccessToken(connector.connectionId);
+  const client = createZendeskClient({
+    token: accessToken,
+    subdomain: configuration.subdomain,
+  });
+  await changeZendeskClientSubdomain({ client, brandId });
+  const categories = await client.helpcenter.categories.list();
+
+  return categories.map((category) => category.id);
```

Comment:
How many categories do we expect to have for big workspaces?

Reply:
not sure, we are gonna need some exploratory work here. Are you anticipating a need for subdividing into multiple activities with pagination here?